homeassistant/components/light/futurenow.py

```diff
@@ -111,25 +111,20 @@ def supported_features(self):
 
     def turn_on(self, **kwargs):
         """Turn the light on."""
-        level = kwargs.get(ATTR_BRIGHTNESS, 255) if self._dimmable else 255
+        if self._dimmable:
+            level = kwargs.get(ATTR_BRIGHTNESS, self._last_brightness)
+        else:
+            level = 255
         self._light.turn_on(to_futurenow_level(level))
-        self._brightness = level
-        self._state = True
-        self._skip_update = True
 
     def turn_off(self, **kwargs):
         """Turn the light off."""
         self._light.turn_off()
-        self._brightness = 0
-        self._state = False
-        self._skip_update = True
+        if self._brightness:
```

Comment:
What if brightness is `0`? Shouldn't it be stored as `last_brightness` then?

Reply:
Brightness 0 is effectively Off state in the dimmer.. Whereas technically it's possible to have state On and brightness 0 this usually happens by some error and the previously set other-than-0 value should be used.